zip-frost.rst

```diff
@@ -0,0 +1,403 @@
+    generated in round one from the ith signer.
+  - sig_share_i, a Scalar value indicating the signature share as produced in
+    round two from the ith signer.
+  - commitment_list =
+      [(j, hiding_nonce_commitment_j, binding_nonce_commitment_j), ...], a
+    list of commitments issued in Round 1 by each signer, where each element
+    in the list indicates the signer identifier j and their two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).s
+    This list MUST be sorted in ascending order by signer identifier.
+  - group_public_key, public key corresponding to the group signing key,
+    an Element in G.
+  - msg, the message to be signed.
+  - randomizer, the randomizer Scalar.
+
+  Outputs: True if the signature share is valid, and False otherwise.
+
+  def verify_signature_share(identifier, PK_i, comm_i, sig_share_i, commitment_list,
+                             group_public_key, msg, randomizer):
+    # Compute the randomized group public key
+    randomized_group_public_key = randomizer * group_public_key
+
+    # Encode the commitment list
+    encoded_commitments = encode_group_commitment_list(commitment_list)
+
+    # Compute the binding factor
+    binding_factor = compute_binding_factor(encoded_commitments, msg, randomizer)
+
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor)
+
+    # Compute the commitment share
+    (hiding_nonce_commitment, binding_nonce_commitment) = comm_i
+    comm_share = hiding_nonce_commitment + (binding_nonce_commitment * binding_factor)
+
+    # Compute the challenge
+    challenge = compute_challenge(group_commitment, randomized_group_public_key, msg)
+
+    # Compute Lagrange coefficient
+    participant_list = participants_from_commitment_list(commitment_list)
+    lambda_i = derive_lagrange_coefficient(identifier, participant_list)
+
+    # Compute relation values
+    l = G.ScalarBaseMult(sig_share_i)
+    r = comm_share + ((challenge * lambda_i) * PK_i)
+
+    return l == r
+
+The `aggregate` function is changed to incorporate the randomizer as follows: ::
+
+  Inputs:
+  - group_commitment, the group commitment returned by compute_group_commitment,
+    an Element in G.
+  - sig_shares, a set of signature shares z_i, Scalar values, for each signer,
+    of length NUM_SIGNERS, where MIN_SIGNERS <= NUM_SIGNERS <= MAX_SIGNERS.
+  - group_public_key, public key corresponding to the group signing key,
+  - challenge, the challenge returned by compute_challenge, a Scalar.
+  - randomizer, the randomizer Scalar.
+
+  Outputs:
+  - (R, z), a Schnorr signature consisting of an Element R and Scalar z.
+  - randomized_group_public_key, the randomized group public key
+
+  def aggregate(group_commitment, sig_shares, group_public_key, challenge, randomizer):
+    randomized_group_public_key = randomizer * group_public_key
```

Comment:
randomizer * G + group_public_key

Reply:
Fixed in 54dbd57d768eadd9d7ee44d79869fc1d3f691f67